### PR TITLE
interaction implemented

### PR DESCRIPTION
**Pull request by ZahraArshia**
Hello 😊 
## in this step of project:
- structure of files is changed to have a "modules" folder related to index.js
- a new "interaction.js" file is added, including methods related to the status updates and reset, so 🔄 and `clear all completed` buttons are now working

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -13,4 +14,7 @@ editbtn.forEach((btn, index) => {
   btn.addEventListener('click', () => {
     newTask.edit(index);
   });
-});
\ No newline at end of file
+});
```
Comment (medaminedev66) [inline src/index.js]: - [ ] You have done a great job by making the task description updated in the localStorage after editing the task description 👍. However, the checkbox status also should be updated in the localStorage after checking/unchecking the checkbox. If the checkbox is checked, the  `completed` value should be true. If the checkbox is unchecked, the `completed` should be false.

File: src/modules/interaction.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+      localStorage.setItem('taskList', JSON.stringify(taskList));
+    }
+  });
+  taskList = taskList.filter((task) => task.completed === false);
+  taskList.forEach((task, id) => {
+    task.index = id + 1;
+  });
+  localStorage.setItem('taskList', JSON.stringify(taskList));
+  location.reload();
+}
```
Comment (medaminedev66) [inline src/modules/interaction.js:24]: - [ ] The functionality of the `clear` and `update` functions doesn't work as expected. Watch this [video ](https://www.loom.com/share/0232fe099ee14207a46b87379547e319) to know more about the issue and kindly work on that.

**Review by medaminedev66 — CHANGES_REQUESTED** [review]
Hi @ZahraArshia , :wave:

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️

_See the inline comments below for more required changes_

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/modules/interaction.js (hunk 2 above)
Comment (ZahraArshia) [reply src/modules/interaction.js:24]: Thanks for the video, I would never find this bug if you didn't mention it. ❤️

**Comment by ZahraArshia** [comment]
Hello 😊 
## in this update:
- "set status" and "get status" functions added to set and keep the value of check box update to tasks `completed `key.
- the bug with editing mode fixed(in previous version it was possible to start another edit in the middle of editing a task, this way the previous editing task would get a null value as description). In this version, edit buttons are disabled during an edit, so the user needs to approve an edit before starting another one.

**Review by medaminedev66 — APPROVED** [review]
## Hi @ZahraArshia 😃 ,

# Status: Approved 

Your project is complete!  There is nothing else to say other than... it's time to merge it 🚢
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**